data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/ownership/ConfigBucketOwnerProviderFactory.java

```diff
@@ -43,9 +52,12 @@ else if(s3SourceConfig.getSqsOptions() != null)
             accountId = extractQueueAccountId(s3SourceConfig);
         else if(s3SourceConfig.getAwsAuthenticationOptions() != null && s3SourceConfig.getAwsAuthenticationOptions().getAwsStsRoleArn() != null)
             accountId = extractStsRoleArnAccountId(s3SourceConfig);
-        else
-            throw new InvalidPluginConfigurationException(
-                    "The S3 source is unable to determine a bucket owner. Configure the default_bucket_owner for the account Id that owns the bucket. You may also want to configure bucket_owners if you read from S3 buckets in different accounts.");
+        else {
+            accountId = defaultAwsCredentialsProvider.resolveCredentials().accountId()
```

Comment:
I understand this is for the L2 transform to pass without a role. @dlvenable Do you think using default credentials account id is ok here?